app/lib/dfe_data_tables/data_elements_loader.rb

```diff
@@ -66,7 +64,11 @@ def process
     end
 
     def concept
-      @concept ||= Concept.find_or_create_by(name: 'No Concept', category: category)
+      return @concept if @concept.present?
+
+      @concept = Concept.find_or_create_by(name: 'No Concept', category: category)
+      @concept.update(description: 'No Description')
```

Comment:
This might be more appropriate in a view helper, as here we're using 'No Description' to mean `nil`, and this will trigger an additional SQL update.

If the logic stands, it might be worth passing a block to `find_or_create_by`, e.g.

```ruby
@concept = Concept.find_or_create_by(name: 'No Concept', category: category) do |new_concept|
  new_concept.description = 'No Description'
end
```

I've not checked this, but I _believe_ it'll reduce it to a single insert.

Reply:
Looking at other parts of this PR, I believe setting the description will break (e.g.) app/views/categories/_concept.html.erb because `description` will always be set, so will not fall back to the placeholder.

Reply:
This will set the 'No Description' placeholder only for the 'No Concept' `Concept`, while it won't affect the other `Concept`s loaded from the Excel file.
Setting this to 'No Description' is in fact necessary to prevent the fallback description to be calculated (over a few thousand `DataElement` records) for the 'No Concept' `Concept`

Reply:
Also, you're right, the block is much more efficient, fixed.